src/videodownlinkbandwidthpolicy/AllHighestVideoBandwidthPolicy.ts

```diff
@@ -23,10 +27,17 @@ export default class AllHighestVideoBandwidthPolicy implements VideoDownlinkBand
   reset(): void {
     this.optimalReceiveSet = new DefaultVideoStreamIdSet();
     this.subscribedReceiveSet = new DefaultVideoStreamIdSet();
+    this.videoSources = [];
+    this.areVideoSourcesChosen = false;
   }
 
   updateIndex(videoIndex: VideoStreamIndex): void {
-    this.optimalReceiveSet = this.calculateOptimalReceiveSet(videoIndex);
+    this.videoIndex = videoIndex;
+    if (!this.areVideoSourcesChosen) {
+      this.optimalReceiveSet = this.calculateOptimalReceiveSet(videoIndex);
+    } else {
+      this.chooseRemoteVideoSources(this.videoSources);
```

Comment:
Instead of this, can we:
- only set videoSources in `chooseRemoteVideoSources`
- in `calculateOptimalreceiveSet`, calculate the set and factor in videoSources.

Reply:
Good point. Let me give it a try. I feel that approach should also work and might be cleaner even.